### ISSUE TITLE
Dockerized solution to obtain localAuthToken 

### ISSUE THREAD

**Issue by sjbeskur** [rithvikvibhu/GHLocalApi#58]
Hi @rithvikvibhu,
Your work on this has been excellent and very insightful. In my attempts to understand how to connect and use the chromecast/ google home API I created a docker image to install prereqs and work through your examples. The repo may be obtained here: https://github.com/advanced-data-machines/chromeblast. Per your excellent work, I am able to successfully obtain the Master token then use this to get the Access Token, then (if the device is registered) use the Access Token to get the Local-Auth-Token (required by all of the POST actions).

Even with the Local-Auth-Token set as the value for "cast-local-authentication-token" I remain unable to issue successful queries to any rest methods that require the "cast-local-authorization-token" in the header? I suspect I am missing something obvious but not sure what else to try. Any help would be greatly appreciated.

**Comment by rithvikvibhu** [comment]
The code in the readme of that repo uses port 8008 with https. That's probably the reason. Try port 8443.

Other than the port change, the last curl at the end should work.

**Comment by sjbeskur** [comment]
I've updated the to use port to 8443 but still get 

```
curl -H 'cast-local-authorization-token: REPLACE_WITH_LOCAL_AUTH_TOKEN' --verbose --insecure https://<ip_of_chromecast>:8443/setup/bluetooth/status
```

Result is always: 
```
*   Trying 10.80.4.35...
* TCP_NODELAY set
* Connected to 10.80.4.35 (10.80.4.35) port 8443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-CHACHA20-POLY1305
* ALPN, server did not agree to a protocol
* Server certificate:
*  subject: C=US; ST=California; L=Mountain View; O=Google Inc; OU=Cast; CN=ADVU275 FA8FCA5E2E12
*  start date: Mar  5 08:00:00 2019 GMT
*  expire date: May 12 12:31:15 2040 GMT
*  issuer: C=US; ST=California; L=Mountain View; O=Google Inc; OU=Cast; CN=Chromecast ICA 12
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
> GET /setup/bluetooth/status HTTP/1.1
> Host: 10.80.4.35:8443
> User-Agent: curl/7.58.0
> Accept: */*
> cast-local-authorization-token: uUDS7QMAAAAAAAAAT3DQwNkteG44u5AAap4Pt1IwKmspqVE6zcQjA1TPgG5oMiT/496XOgBzQh4zEqkU1wmS871i+L4tZ1KMJrdqUOSTzng=
> 
< HTTP/1.1 404 Not Found
< Content-Length:0
< 
* Connection #0 to host 10.80.4.35 left intact
```

I notice in your get_tokens.sh there is a reference to device_id:

```python
if not device_id:
    device_id = _get_android_id()
```
Is this script intended to work only on Android?

**Comment by rithvikvibhu** [comment]
Hey, so I just tried with my Google Home, and seems to work fine.
```
curl https://192.168.0.23:8443/setup/bluetooth/status -H "cast-local-authorization-token: TOKEN" --verbose -k
```

Just to confirm, you're trying with a Google Home, right? Even if a Chromecast uses a similar format, it doesn't support everything a Google Home does, Bluetooth being one of the features (afaik).

To see if the token is correct, try sending a random token. If you get Forbidden, then there's no problem with the token.
In this case, the response is 404, which is why I asked if it's a Google Home.

**Comment by sjbeskur** [comment]
@rithvikvibhu Thank you for your response... it makes perfect sense.   I am indeed only trying to configure a Chromecast and had thought that I could do this through Google Home API's... that does not seem to be the case.   I came across the following reddit post: https://www.reddit.com/r/Chromecast/comments/bmv2f7/script_to_batch_rename/  which is how I found your project and ended up down this rabbit hole.
